Add unifi-network extension

lewjuh opened the pull request:
## Description

<!-- A summary of your change. If you add a new extension or command, explain what it does. -->

## Screencast

<!-- If you add a new extension or command, include a screencast (or screenshot for straightforward changes). A good screencast will make the review much faster - especially if your extension requires registration in other services.  -->

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

pernielsentikaer reviewed (changes requested):
Hi 👋

Did you see the [UniFi](https://www.raycast.com/tonka3000/unifi) extension before creating this?

lewjuh commented:
> Hi 👋
> 
> 
> 
> Did you see the [UniFi](https://www.raycast.com/tonka3000/unifi) extension before creating this?

Hey! Ah, sadly I did not. This does differ in that it uses the official UniFi API recently released, but understand if dups aren't wanted in the store.

pernielsentikaer commented:
Hi @lewjuh 👋 

I spoke with the author of the other extension, he is willing to let you take over the ownership of current and update it with your code - could you do that instead of opening a new? Feel free to change author too when you do

Do you have any questions?

lewjuh commented:
> Hi @lewjuh 👋
> 
> I spoke with the author of the other extension, he is willing to let you take over the ownership of current and update it with your code - could you do that instead of opening a new? Feel free to change author too when you do
> 
> Do you have any questions?
Hey, thanks for sorting that, I appreciate it!

Happy to do that, is there any concern about people who have the old version installed? How does the updating work on raycast extensions for a user?

pernielsentikaer commented:
Let's see if we can come up with a way. We should follow the best approach by using an API key, as you're doing, instead of relying on a username/password. We could make the new preference required and remove the two other preferences; that will force the user to add the key.

Another option is to set the new preference as optional and show a special view to let the user know about the change and guide them a bit better on how to do that.

The second approach could be a cool thing, but it might be overkill.

lewjuh commented:
@pernielsentikaer Will look at doing the latter, to try make it as easy as possible!

Have updated and refactored this for now as needed some extra function.

pernielsentikaer commented:
So is this already ready for review?

lewjuh commented:
> So is this already ready for review?

Sorry nope, just refactored and added to my code, i need to look at the other extension to see how best to integrate it.

pernielsentikaer commented:
Awesome, thanks for checking. Since we're taking it over, it's nice if we can do it as softly as possible 😊

lewjuh commented:
@pernielsentikaer Hey mate, wasnt sure how exactly to go about updating the other one but just went and directly updated it.  #17101 

Let me know if thats okay, i went for the option of prompting to update the user but still having the legacy stuff work. Can probably close this now and move over to their to review.

pernielsentikaer commented:
Closing this in favor of https://github.com/raycast/extensions/pull/17101
